gateway/api.go

```diff
@@ -3268,3 +3271,21 @@ func updateOASServers(spec *APISpec, conf config.Config, apiDef *apidef.APIDefin
 	newAPIURL := getAPIURL(*apiDef, conf)
 	oasObj.UpdateServers(newAPIURL, oldAPIURL)
 }
+
+// GetSessionLifetime defines the TTL of the key
+func GetSessionLifetime(sess *user.SessionState, gw *Gateway) int64 {
+	var lifetime int64
+	if sess != nil && len(sess.AccessRights) > 0 {
```

Comment:
we need to check if we are doing this in other parts of the gw code - maybe we can reutilize

Reply:
done! normalized the behavior